sql/hive/src/test/scala/org/apache/spark/sql/hive/InsertSuite.scala

```diff
@@ -927,4 +927,23 @@ class InsertSuite extends QueryTest with TestHiveSingleton with BeforeAndAfter
       testDefaultColumn
     }
   }
+
+  test("SPARK-50469: V1Writes should respect the output ordering") {
+    withTable("t1") {
+      withTempView("t2") {
+        val df = scala.util.Random.shuffle((1 to 1000).map(i => (i, "part_val"))).toDF("key", "val")
+        df.createOrReplaceTempView("t2")
+
+        sql("CREATE TABLE t1 (key Int, part String) PARTITIONED BY (part)")
```

Comment:
shall we add `USING hive` to explicitly test hive tables?

Reply:
But do we really need to test hive? The `V1Write` framework isn't really related to the catalog.

Reply:
`InsertIntoHiveTable` is also `V1Write`, will update this test.